freqtrade/freqtradebot.py

```diff
@@ -55,7 +55,15 @@ def __init__(self, config: Dict[str, Any]) -> None:
         self.strategy: IStrategy = StrategyResolver(self.config).strategy
 
         self.rpc: RPCManager = RPCManager(self)
-        self.exchange = Exchange(self.config)
+
+        exchange_name = self.config.get('exchange', {}).get('name', 'bittrex').title()
+        try:
+            self.exchange = ExchangeResolver(exchange_name, self.config).exchange
+        except ImportError:
```

Comment:
One additional change if you don't mind:
i will move handling this error to the Resolver (so it returns either type Kraken, or falls back to type Exchange).

that will make it easier to work with (and should make dynamic tests easier i hope).

Reply:
Yes please, that's something I was thinking about as well.